MediaGalleryUi/view/adminhtml/ui_component/media_gallery_listing.xml

```diff
@@ -35,6 +36,7 @@
         </dataProvider>
     </dataSource>
     <listingToolbar name="listing_top">
+        <filterSearch name="keywords"/>
```

Comment:
I think it could be more optimal to reuse the out of the box fulltext type. Should work without any changes to data provider and filter classes. Also it would be good if the search will consider not only keywords but all the text fields in the grid table (that should also work by default).
```suggestion
        <filterSearch name="fulltext"/>
```

Reply:
@sivaschenko to add the default full-text search we need to add the full-text index for desired columns. This can be done in 2 ways: mview or general full-text index. I am not totally sure about this. Please, check.